sdk/provisioning/Azure.Provisioning.Sql/tests/BasicSqlTests.cs

```diff
@@ -66,16 +58,16 @@ await test.Define(
             @description('The name of the SQL Database.')
             param dbName string = 'SampleDB'
 
-            @description('The SQL Server location.')
-            param location string = resourceGroup().location
-
             @description('The administrator username of the SQL logical server.')
             param adminLogin string
 
             @secure()
             @description('The administrator password of the SQL logical server.')
             param adminPass string
 
+            @description('The location for the resource(s) to be deployed.')
+            param location string = resourceGroup().location
```

Comment:
(nit) would it be possible to generate this parameter first? That's how the old CDK did it. It kind of reads better IMO.

Reply:
I don't think it's worth it.  Right now we leave params in the order you defined them, just all shifted up top.  If we started reordering them it could make anyone using explicit `location` params confused that we're shuffling everything on them.  I want to leave this kind of dumb in Azure.Provisioning, but you could write your own resolver that does it cheaply.  You'd just want to special case that first condition on a `StartsWith("location")`: https://github.com/Azure/azure-sdk-for-net/blob/aae6107fb8af07de62cb73fc96ae4f5d04609eab/sdk/provisioning/Azure.Provisioning/src/Primitives/OrderingInfrastructureResolver.cs#L16-L25